Modules/hmacmodule.c

```diff
@@ -0,0 +1,277 @@
+#ifndef Py_BUILD_CORE_BUILTIN
+#  define Py_BUILD_CORE_MODULE 1
+#endif
+
+#include "pyconfig.h"
+#include "Python.h"
+#include "hashlib.h"
+
+#include "_hacl/Hacl_HMAC.h"
+
+#include "clinic/hmacmodule.c.h"
+
+#define HACL_HMAC_COMPUTE_HASH(HACL_FUNCTION, DIGEST_SIZE, KEY, SRC)    \
+    do {                                                                \
+        unsigned char out[DIGEST_SIZE];                                 \
+        HACL_FUNCTION(                                                  \
+            out,                                                        \
+            (uint8_t *)KEY->buf, (uint32_t)KEY->len,                    \
+            (uint8_t *)SRC->buf, (uint32_t)SRC->len                     \
+        );                                                              \
+        return PyBytes_FromString((const char *)out);                   \
+    } while (0)
+
+/*[clinic input]
+module _hmac
+[clinic start generated code]*/
+/*[clinic end generated code: output=da39a3ee5e6b4b0d input=799f0f10157d561f]*/
+
+/*[clinic input]
+@critical_section
```

Comment:
I know very little about the library here, but do we really need critical sections? The lock contention here is going to suck when used in parallel. (I would really hope that HACL * doesn't need global state protected by a lock.)

Reply:
no such thing as global state in HACL

Reply:
> I know very little about the library here, but do we really need critical sections? The lock contention here is going to suck when used in parallel

Actually, you're probably right for this one. I don't think I need a critical section here since it's a one-shot.